File: backend/src/main/java/vn/gtel/pm2/sizing/dto/response/ProjectResponse.java
package vn.gtel.pm2.sizing.dto.response;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProjectResponse {
    private Long id;
    private UserResponse owner;
    private String name;
    private String description;
    private List<CatalogComponentResponse> selectedCatalogComponents;
    private ProjectAssumptionResponse projectAssumption;
    private Instant createdAt;
    private Instant updatedAt;
    private UUID createdBy;
    private UUID updatedBy;
}

File: frontend/src/features/project/types/response/project-response.ts
import { CatalogComponentResponse } from "@/features/catalog-component/types/response/catalog-component-response";
import type { ComponentSizingResultResponse } from "./component-sizing-result-response";
import type { ProjectAssumptionResponse } from "./project-assumption-response";
import type { ProjectOwnerResponse } from "./project-owner-response";

export type ProjectResponse = {
  id: number;
  owner: ProjectOwnerResponse;
  name: string;
  description: string;
  selectedCatalogComponents: CatalogComponentResponse[];
  projectAssumption: ProjectAssumptionResponse | null;
  sizingResults: ComponentSizingResultResponse[];
  totalMachinesResult: number;
  createdAt: string;
  updatedAt: string;
  createdBy: string | null;
  updatedBy: string | null;
};
